### PR TITLE
DP-2574: doc_dependencies script

### PR DESCRIPTION
**Pull request by Mara3l**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: scripts/bump_doc_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    # Write the updated data back to the TOML file
+    try:
+        with open(file_path, "w") as file:
+            file.write(tomlkit.dumps(data))
+    except Exception as e:
+        print("Error updating params.toml:", e)
+
+
+def bump_redir(file_path, new_version):
```
Comment (hkad98) [inline scripts/bump_doc_dependencies.py:46]: Missing types.

File: scripts/bump_doc_dependencies.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    new_version_data["dirpath"] = new_version
+    new_version_data["url"] = f"/{new_version}/"
+    versions.insert(1, new_version_data)
+
+    # Write the updated data back to the TOML file
+    try:
+        with open(file_path, "w") as file:
+            file.write(tomlkit.dumps(data))
+    except Exception as e:
+        print("Error updating params.toml:", e)
```
Comment (hkad98) [inline scripts/bump_doc_dependencies.py:43]: I think this is not necessary.

File: scripts/bump_doc_dependencies.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def bump_redir(file_path, new_version):
+    short_version = ".".join(new_version.split(".")[:2])
+
+    try:
+        new_redirect_rule = f"/{new_version}/ {{{{ .Site.BaseURL }}}}/{short_version} 301!\n"
+        with open(file_path, "r") as file:
+            content = file.readlines()
```
Comment (hkad98) [inline scripts/bump_doc_dependencies.py:52]: Why do you read and write redir multiple times in the function? Could you rewrite it so it is used only once?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by hkad98 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR317

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul/131/ : FAILURE in 1m 15s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/281/ : SUCCESS in 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/284/ : SUCCESS in 10s

**Pull request merged**